src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -263,15 +265,15 @@ synchronized int insertRow(String tableName, List<Object> row) throws SQLExcepti
   }
 
   @Override
-  public synchronized <T> @org.checkerframework.checker.nullness.qual.Nullable List<T> readLatestGroup(Class<T> clz)
+  public synchronized <T, E> @org.checkerframework.checker.nullness.qual.Nullable List<T> readAllForMaxField(Class<T> clz, Field<E> field)
           throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException, DataAccessException {
     String tableName = getTableNameFromClassName(clz);
-    Field<String> actionName = DSL.field(PersistedAction.SQL_SCHEMA_CONSTANTS.ACTION_COL_NAME, String.class);
     List<Record> maxTimeStampRecordList;
 
     try {
       // Fetch the latest rows with the last timestamp.
-      maxTimeStampRecordList = create.select().from(tableName).groupBy(actionName).fetch();
+      maxTimeStampRecordList = create.select().from(tableName).where(DSL.field(field)
```

Comment:
Since you are trying to keep this fn generic (not specific to latest timestamp), this variable should also not be timestamp specific. Maybe something on the lines of `recordsWithMaxFieldValue`

Reply:
We should also test the generality by adding a UT for some other (non-timestamp) field and checking that we get all fields with max value.
FWIW, I think it's also ok to make this function only timestamp specific and get all records with the latest timestamp value instead of a generic max-field value.

Reply:
Made the API generic to address future use-cases whatsoever. Added the UTs as well.